include/zephyr/arch/arm/cortex_a_r/cpu.h

```diff
@@ -26,6 +26,12 @@
 #define MODE_SYS	0x1f
 #define MODE_MASK	0x1f
 
+#if defined(CONFIG_BIG_ENDIAN)
+#define E_BIT   (1 << 9)
+#else
+#define E_BIT   (0 << 9)
+#endif
+
```

Comment:
Set or un-set `E_BIT` in `thread.c`. This file is encoding the BIT position, so the `E_BIT` position is always `1 << 9` but only set for `CONFIG_BIG_ENDIAN`.